### PR TITLE
Set default AppxBundlePlatforms

### PR DESCRIPTION
**Pull request by NickGerleman**
### Type of Change
- Bug fix (non-breaking change which fixes an issue)

### Why
The `AppxBundlePlatforms` property determines which platforms are added to the generated AppX. AppX target logic to detemine a default for this is a bit suspect, and will force a rebuild of the project in any architectures not already built.

### What
This change makes sure we only build a single platform at a time for our debug builds, and makes sure we build ARM64 by default for release builds.

To keep CI parallel, I've added an env setup step to only build a single platform at a time in PRs. I did not change CI logic, in order to emulate a more stock configuration. This means each release build in CI will however build all three platforms I think, and this may require some adjustments. Needed to clean up our CI environment init code a bit to apply this eveywhere it should go.

## Testing
Will validate expected build times and steps run when running PR.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/9006)

### CODE REVIEW

**Comment by NickGerleman** [comment]
Looks like we're building the right things in PR:

image

File: vnext/PropertySheets/Appx.props [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <PropertyGroup>
+    <_NormalizedPlatform Condition="'$(Platform)' == 'Win32'">x86</_NormalizedPlatform>
+    <_NormalizedPlatform Condition="'$(Platform)' != 'Win32'">$(Platform)</_NormalizedPlatform>
+  </PropertyGroup>
+
+  <!--
```
Comment (dannyvv) [inline vnext/PropertySheets/Appx.props:9]: ❤️

**Review by dannyvv — APPROVED** [review]

File: vnext/PropertySheets/Appx.props [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    AppX bundling logic has a bug causing it to rebuild the project as x64 when
+    specifying x86 configuration. Only build the configuration we asked for in
+    debug builds, but build everything by default in release builds.
+  -->
+  <PropertyGroup Condition="'$(AppxBundlePlatforms)' == ''">
+    <AppxBundlePlatforms Condition="'$(Configuration)' == 'Debug'">$(_NormalizedPlatform)</AppxBundlePlatforms>
+    <AppxBundlePlatforms Condition="'$(Configuration)' == 'Release'">x86|x64|arm64</AppxBundlePlatforms>
```
Comment (NickGerleman) [inline vnext/PropertySheets/Appx.props:16]: Testing locally that this works as expected, since we will not get PR coverage.

**Comment by NickGerleman** [comment]
@msftbot merge in 96 hours

**Comment by NickGerleman** [comment]
Merging this (and some other PRs) to avoid conflicts.

**Pull request merged**